sympy/functions/special/gamma_functions.py

```diff
@@ -133,6 +162,11 @@ def eval(cls, arg):
                     else:
                         return 2**n*sqrt(S.Pi) / coeff
 
+        z = arg.extract_multiplicatively(S.ImaginaryUnit)
+        if z is not None:
+            if z is S.Infinity or S.NegativeInfinity:
```

Comment:
`if z is S.Infinity or z is S.NegativeInfinity` we need `z is S.NegativeInfinity`, not just `S.NegativeInfinity` otherwise it will be `True` for any non-None `z`.

Also we wouldn't need `if z is not None` I think.